test/tools/dshell.d

```diff
@@ -0,0 +1,171 @@
+/**
+A small library to help write D shell-like test scripts.
+*/
+module dshell;
+
+public import std.string;
+public import std.format;
+public import std.path;
+public import std.file;
+public import std.stdio;
+public import std.process;
+
+struct TestEnv
+{
+    string MODEL;
+    string RESULTS_DIR;
+    string EXE;
+    string OBJ;
+    string DMD;
+    string DFLAGS;
+    string OS;
+    string SEP;
+    string DSEP;
+}
+__gshared TestEnv env;
+__gshared string TEST_DIR; // TEST_DIR should be one of compilable, fail_compilation or runnable
```

Comment:
> TEST_DIR should be one of compilable, fail_compilation or runnable

So why not using a named `enum TEST_DIR : string {}` ?

Reply:
You could, but I'm not seeing the benefit...?

Reply:
I suppose that would let you do a final switch on it?

Though, I don't see anyone doing that with TEST_DIR...but I would see people doing that with other values...such as OS

Reply:
No it was just a question of principle. The initial comment was simply motivated by the fact that you seemed to say that the variable could only have 3 different values. So just forget this, this was not a request.